cal-automation-server-recording/api/controllers/CallRecordingController.ts

```diff
@@ -0,0 +1,435 @@
+import {
+  ServerCallLocator,
+  StartRecordingOptions,
+  RecordingChannel,
+  CallRecording
+} from "@azure/communication-call-automation";
+import { Request, Response } from "express";
+import * as fs from "fs";
+import Root from "../../Root";
+import { Mapper, FileFormat, FileDownloadType } from "../../FileFormat";
+import { BlobDownloadResponseModel } from "@azure/storage-blob";
+
+var cfg = require("../../config");
+
+const connectionString = cfg.ConnectionString;
+const callbackUri = cfg.CallbackUri;
+
+var CallAutomationClient = require("@azure/communication-call-automation");
+const { EventGridDeserializer } = require("@azure/eventgrid");
+const { BlobStorageHelper } = require("../../BlobStorageHelper");
+const { Logger, MessageType } = require("../../Logger");
+
+const client = new CallAutomationClient.CallAutomationClient(connectionString)
+
+let recordingData = new Map<string, string>();
+let recFileFormat: FileFormat = FileFormat.mp4;
+
+exports.startUp = function (req: Request, res: Response) {
+  res.json("App is running...");
+};
+
+exports.startRecording = async function (req: Request, res: Response) {
+  try {
+    let serverCallId: string = req.query.serverCallId!.toString();
+    if (!serverCallId || String(serverCallId).trim() == "") {
+      return res.status(400).json("serverCallId is invalid");
+    }
+    Logger.logMessage(
+      MessageType.INFORMATION,
+      "Start recording API called with serverCallId =  " + serverCallId
+    );
+    var locator: ServerCallLocator = { id: serverCallId };
+    var options:StartRecordingOptions={callLocator:locator}
+    var output = await client.getCallRecording().startRecording(options)
+
+    recordingData.set(serverCallId, output.recordingId);
+    return res.json(output);
+  } catch (e) {
+    var output = BlobStorageHelper.getExecptionDetails(e);
+    return res.status(output.statusCode).json(String(output.output));
+  }
+};
+
+exports.startRecordingWithOptions = async function (req: Request, res: Response) {
+  try {
+    let serverCallId: string = req.query.serverCallId!.toString();
+
+    if (!serverCallId || String(serverCallId).trim() == "") {
+      return res.status(400).json("serverCallId is invalid");
+    }
+
+    let recordingContent: string = req.query.recordingContent
```

Comment:
These can be better organized, in one pass, rather 2 pass.

Reply:
Hi @moirf I have worked on this comment .Tried to simplify for single pass but here both checks are needed so anyway 2 checks are definitely required to add in code. I have created a general function for both checks and committed changes.